### PR TITLE
Dispatch on needed branches only

### LINKED ISSUE #115: Dispatch: Do PR only on stable first, then on master
The goal is to reduce the Travis load when the Dispatcher is running.

Most of the case, PR on stable and master branch are the same.

The proposed logic:

1. Check if some diff should be applied on stable branch
2. If yes, make a PR
3. Check if a PR is present for stable branch
4. If not, check if the stable branch is fully merged on master
5. If not and diff are present, make a PR against master

### PR DESCRIPTION
**Pull request by Soullivaneuh**
Closes #115.

### CODE REVIEW

File: src/Console/Command/DispatchCommand.php [hunk 1]
```diff
@@ -207,52 +207,80 @@ private function dispatchFiles(Package $package)
...
+        $previousDevKit = null;
+        while (($branchConfig = current($branches))) {
+            // We have to fetch all branches on each step in case a PR is submitted.
+            $remoteBranches = array_map(function ($branch) {
+                return $branch['name'];
+            }, $this->githubClient->repos()->branches(static::GITHUB_GROUP, $repositoryName));
+
+            $currentBranch = key($branches);
+            $currentDevKit = $currentBranch.'-dev-kit';
+            next($branches);
```
Comment (greg0ire) [inline src/Console/Command/DispatchCommand.php]: Putting this right in the middle of the `while` statement feels a bit weird… why not line 215 ?

File: src/Console/Command/DispatchCommand.php [hunk 2]
```diff
@@ -207,52 +207,80 @@ private function dispatchFiles(Package $package)
...
+
+            // A PR is already here for previous branch, do nothing on the current one.
+            if (in_array($previousDevKit, $remoteBranches, true)) {
+                continue;
+            }
+            // If the previous branch is not merged to the current one, do nothing.
```
Comment (greg0ire) [inline src/Console/Command/DispatchCommand.php]: "merged to" => "merged into"

File: src/Console/Command/DispatchCommand.php (hunk 1 above)
Comment (Soullivaneuh) [inline src/Console/Command/DispatchCommand.php]: Because `key($branches);` would not give the same result after the `next`.

Comment (greg0ire) [inline src/Console/Command/DispatchCommand.php]: Right

**Comment by greg0ire** [comment]
LGTM

**Pull request merged**